mcux/middleware/wifi_nxp/wifidriver/mlan_init.c

```diff
@@ -0,0 +1,567 @@
+/** @file mlan_init.c
+ *
+ *  @brief  This file provides initialization for FW and HW
+ *
+ *  Copyright 2008-2024 NXP
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+
+/********************************************************
+Change log:
+    10/13/2008: initial version
+********************************************************/
+
+#include <mlan_api.h>
+
+/* Additional WMSDK header files */
+#include <wmerrno.h>
+#include <osa.h>
+#include "fsl_common.h"
+
+/* Always keep this include at the end of all include files */
+#include <mlan_remap_mem_operations.h>
+/********************************************************
+        Global Variables
+********************************************************/
+
+#define BOARD_SDMMC_DATA_BUFFER_ALIGN_SIZE 32
+
+//_IOBUFS_ALIGNED(SDIO_DMA_ALIGNMENT)
+#if defined(SD8978) || defined(SD8987) || defined(SD8997) || defined(SD9097) || defined(SD9098) || defined(SD9177)
+static t_u8 mp_regs_buffer[MAX_MP_REGS + DMA_ALIGNMENT];
+#elif defined(SD8801)
+SDK_ALIGN(uint8_t mp_regs_buffer[MAX_MP_REGS], BOARD_SDMMC_DATA_BUFFER_ALIGN_SIZE);
```

Comment:
We may not be able to use the `SDK_ALIGN` declaration here- it is part of the `fsl_common.h` header, but this won't be provided by non-NXP host SOCs. We should probably just provide a definition for an alignment macro within this file, to keep the contents portable between Zephyr and MCUX SDK.